src/App.jsx

```diff
@@ -1,34 +1,122 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { getTodos } from './api/api';
 import './App.scss';
 import './styles/general.scss';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
 
-class App extends React.Component {
-  state = {
-    todos: [],
-    selectedUserId: 0,
+const App = () => {
+  const [todos, setTodos] = useState([]);
+  const [filteredTodos, setFilteredTodos] = useState([]);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [text, addText] = useState('');
+  const [status, setStatus] = useState('All');
+
+  useEffect(() => {
+    getTodos()
+      .then((list) => {
+        const listFromServer
+          = list.data.filter(item => item.title && item.id && item.userId);
+
+        setTodos(listFromServer);
+        setFilteredTodos(listFromServer);
+      });
+  }, []);
+
+  const selectUser = (event) => {
+    setSelectedUserId(event.target.textContent);
   };
 
-  render() {
-    const { todos, selectedUserId } = this.state;
+  const userClear = () => {
+    setSelectedUserId('');
+  };
 
-    return (
-      <div className="App">
-        <div className="App__sidebar">
-          <TodoList todos={todos} />
-        </div>
+  const filterByTitle = (value) => {
+    let result = [];
+
+    addText(value);
+    result = selectStatus(status, todos, value);
+    setFilteredTodos(result);
+  };
+
+  const selectStatus = (currentStatus, arrayTodos = todos, query = text) => {
```

Comment:
you're not only selecting status in this function, you also filtering todos by title. Function name should describe this behavior